Korean National Library.js

```diff
@@ -0,0 +1,578 @@
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "https://www.nl.go.kr/NL/contents/search.do?srchTarget=total&pageNum=1&pageSize=10&kwd=%EB%AF%B8%EA%B5%B0%EC%A0%95%EA%B3%BC+%ED%95%9C%EA%B5%AD%EC%9D%98+%EB%AF%BC%EC%A3%BC%EC%A3%BC%EC%9D%98",
+		"items": "multiple"
+	},
+	{
+		"type": "web",
+		"url": "https://www.nl.go.kr/EN/contents/engSearch.do?resultType=&pageNum=1&pageSize=30&order=&sort=&srchTarget=total&kwd=lawson&systemType=&lnbTypeName=&category=&hanjaFlag=&reSrchFlag=&licYn=&kdcName1s=&manageName=&langName=&ipubYear=&pubyearName=&seShelfCode=&detailSearch=&seriesName=&mediaCode=&offerDbcode2s=&f1=&v1=&f2=&v2=&f3=&v3=&f4=&v4=&and1=&and2=&and3=&and4=&and5=&and6=&and7=&and8=&and9=&and10=&and11=&and12=&isbnOp=&isbnCode=&guCode2=&guCode3=&guCode4=&guCode5=&guCode6=&guCode7=&guCode8=&guCode11=&gu2=&gu7=&gu8=&gu9=&gu10=&gu12=&gu13=&gu14=&gu15=&gu16=&subject=&sYear=&eYear=&sRegDate=&eRegDate=&typeCode=&acConNo=&acConNoSubject=&infoTxt=#viewKey=CNTS-00091570741&viewType=C&category=%EA%B8%B0%EC%82%AC&pageIdx=2&jourId=",
+		"items": [
+			{
+				"itemType": "journalArticle",
+				"title": "THE GROMOV-LAWSON-ROSENBERG CONJECTURE FOR GROUPS WITH PERIODIC COHOMOLOGY",
+				"creators": [
+					{
+						"firstName": "Gilkey P.",
+						"lastName": "Botvinnik B",
+						"creatorType": "author"
+					}
+				],
+				"date": "1997-07-01",
+				"language": "eng",
+				"libraryCatalog": "Korean National Library",
+				"publicationTitle": "Journal of Differential Geometry. V.46 N.3",
+				"rights": "2",
+				"attachments": [],
+				"tags": [],
+				"notes": [
+					{
+						"note": "자료 조회를 위해서는 별도 뷰어를 설치해야 합니다.뷰어 설치: https://www.cartesianinc.com/Products/CPCView/"
+					},
+					{
+						"note": "LG상남도서관 기증 자료"
+					},
+					{
+						"note": "Botvinnik B UNIV OREGON EUGENE, OR 97403 USA  UNIV NOTRE DAME NOTRE DAME, IN 46556 USA"
+					}
+				],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "https://www.nl.go.kr/EN/contents/engSearch.do?resultType=&pageNum=1&pageSize=30&order=&sort=&srchTarget=total&kwd=smith&systemType=%EC%98%A8%EB%9D%BC%EC%9D%B8%EC%9E%90%EB%A3%8C&lnbTypeName=&category=%EB%A9%80%ED%8B%B0%EB%AF%B8%EB%94%94%EC%96%B4&hanjaFlag=&reSrchFlag=&licYn=&kdcName1s=&manageName=&langName=&ipubYear=&pubyearName=&seShelfCode=&detailSearch=&seriesName=&mediaCode=&offerDbcode2s=&f1=&v1=&f2=&v2=&f3=&v3=&f4=&v4=&and1=&and2=&and3=&and4=&and5=&and6=&and7=&and8=&and9=&and10=&and11=&and12=&isbnOp=&isbnCode=&guCode2=&guCode3=&guCode4=&guCode5=&guCode6=&guCode7=&guCode8=&guCode11=&gu2=&gu7=&gu8=&gu9=&gu10=&gu12=&gu13=&gu14=&gu15=&gu16=&subject=&sYear=&eYear=&sRegDate=&eRegDate=&typeCode=&acConNo=&acConNoSubject=&infoTxt=#viewKey=CNTS-00098977147&viewType=C&category=%EB%A9%80%ED%8B%B0%EB%AF%B8%EB%94%94%EC%96%B4&pageIdx=17&jourId=",
+		"items": [
+			{
+				"itemType": "videoRecording",
+				"title": "두 단어 영어로 쉽게 대답하기 step 1. 별일 없이 지내. / Smith씨하고 통화할 수 있을까요?",
+				"creators": [
+					{
+						"firstName": "",
+						"lastName": "조정현",
```

Comment:
@kmlawson could you help us on that?

Reply:
This is indeed tricky. In this specific example, you have a typical Korean name: 조정현 <- Any Korean person looking at that will immediately recognize that 조 is the last name, and 정현 is the first name. You can probably cover some 90% or more of all Korean authors if you were able to code the following logic flow:

1)  Is the author in 2-byte Korean hangul OR in Chinese characters (there must be standard libraries for this, recognizing characters as one of a subset of the Unicode set)

2) If so, are there either 2 or 3 characters (this reduces the possibility you are looking at a Japanese name)

3) If so, grab the first character (Hangul Korean letter compound such as 김 or Hanja/Hanzi Chinese character equivalent 金) and assign it to last name. Assign the remaining one or two characters to the first name.

4) Else, just default to dumping it all in last name and let the user sort it out if there is no indication in the metadata that they are split.

This will cover not all, but the vast majority of cases, including all the test cases here, including the more rare two character name 안진 example above (that is weird case though: their full name may be 유안진 when I look them up, but they only put 안진 as their author, which should be their first name - maybe a pen name - but in any case, the vast majority of rare two character names will be one character last name and one character first name).

Cases that are harder:
1) There are lots of Japanese authors in the national library, obviously especially from the colonial period. Japanese names have much more variety, with two characters last name being, by far, the most common. However, it won't be easy to know if you are looking at a Chinese name, a Korean name written in Chinese characters, or a Japanese name if there are only 3 characters. More than 3 will be Japanese name or some kind of institutional author.

Looks like MARC data does *not* separate first/last name.

Example Book: 鐵道運轉取扱理論と運用 by 鈴木東司 <- last name 鈴木 (Suzuki)
https://www.nl.go.kr/EN/contents/engSearch.do?resultType=&pageNum=1&pageSize=30&order=&sort=&srchTarget=total&kwd=%E9%88%B4%E6%9C%A8&systemType=&lnbTypeName=&category=%EB%8F%84%EC%84%9C&hanjaFlag=&reSrchFlag=&licYn=&kdcName1s=&manageName=&langName=&ipubYear=1899_1947&pubyearName=&seShelfCode=&detailSearch=&seriesName=&mediaCode=&offerDbcode2s=&f1=&v1=&f2=&v2=&f3=&v3=&f4=&v4=&and1=&and2=&and3=&and4=&and5=&and6=&and7=&and8=&and9=&and10=&and11=&and12=&isbnOp=&isbnCode=&guCode2=&guCode3=&guCode4=&guCode5=&guCode6=&guCode7=&guCode8=&guCode11=&gu2=&gu7=&gu8=&gu9=&gu10=&gu12=&gu13=&gu14=&gu15=&gu16=&subject=&sYear=&eYear=&sRegDate=&eRegDate=&typeCode=&acConNo=&acConNoSubject=&infoTxt=#viewKey=6061103&viewType=AH1&category=%EB%8F%84%EC%84%9C&pageIdx=21&jourId=


2) Western names. They appear to be left in English. Looks like MARC data separates first/last. 

Example book: North Korea: Another Country by Bruce Cumings  
https://www.nl.go.kr/EN/contents/engSearch.do?resultType=&pageNum=1&pageSize=10&order=&sort=&srchTarget=total&kwd=bruce+cumings+korea&systemType=&lnbTypeName=&category=%EB%8F%84%EC%84%9C&hanjaFlag=&reSrchFlag=&licYn=&kdcName1s=&manageName=&langName=&ipubYear=&pubyearName=&seShelfCode=&detailSearch=&seriesName=&mediaCode=&offerDbcode2s=&f1=&v1=&f2=&v2=&f3=&v3=&f4=&v4=&and1=&and2=&and3=&and4=&and5=&and6=&and7=&and8=&and9=&and10=&and11=&and12=&isbnOp=&isbnCode=&guCode2=&guCode3=&guCode4=&guCode5=&guCode6=&guCode7=&guCode8=&guCode11=&gu2=&gu7=&gu8=&gu9=&gu10=&gu12=&gu13=&gu14=&gu15=&gu16=&subject=&sYear=&eYear=&sRegDate=&eRegDate=&typeCode=&acConNo=&acConNoSubject=&infoTxt=#viewKey=65838543&viewType=AH1&category=%EB%8F%84%EC%84%9C&pageIdx=4&jourId=

Reply:
I'll leave this for Zotero 7. The js regex built into the current Zotero (Firefox ~60) engine is missing some unicode functionality I'd like to have for this that should be available in Firefox >77 (and thus Zotero 7)

Reply:
Do you know about `ZU.XRegExp()`?

https://github.com/zotero/translators/blob/fc2a1caf28a1a78eec943cb2ec58d76bbff37a9f/BibLaTeX.js#L305

Reply:
Thank Dan, I knew in theory we had this, but had forgotten.
@kmlawson -- so I'm actually able to regex parse against unicode script names:
https://en.wikipedia.org/wiki/Script_(Unicode)#List_of_scripts_in_Unicode

If you could just confirm I'm understanding this correctly:
> Is the author in 2-byte Korean hangul OR in Chinese characters 

That'd translate to `\p{Korean}` -- i.e. either Han or Hangul

So to make sure I understand the above correctly, I'd do something like this (wrriten in pseudo-code)

```
for creator of creators --
  if (creator is single field)
    len = creator.length
    if len >3 
       continue
    else
       if (/\p{Korean}{len}/.test(creator))
         - first letter is lastname, rest is firstname
```  

This will work reliably for Korean authors and Western authors, but won't work for Japanese authors, which it will in some cases split incorrectly, in others leave in single field. I think that's perfectly acceptable (not let perfect be enemy of the good etc.)

Reply:
Yes, that looks right! If length greater than 3, dump it all in last name, else, first letter (2-byte compound character, either Korean or Chinese characters  - looks like {Korean} includes both hangul and hanja Chinese characters) is last name and the rest first (in the vast majority of cases).